iOS/Flows/Home/_New/new_ConversationViewController.swift

```diff
@@ -82,76 +99,56 @@ class new_ConversationViewController: FullScreenViewController, CollectionViewIn
             self.conversationController.deleteMessage(message.id)
         }
     }
-
-    override func viewDidLayoutSubviews() {
-        super.viewDidLayoutSubviews()
-
-        self.blurView.expandToSuperviewSize()
-        self.collectionView.expandToSuperviewSize()
-    }
 }
 
 extension new_ConversationViewController: UICollectionViewDelegate {
 
+
     func collectionView(_ collectionView: UICollectionView,
                         willDisplay cell: UICollectionViewCell,
                         forItemAt indexPath: IndexPath) {
 
         // If all the messages are loaded, no need to fetch more.
         guard !self.conversationController.hasLoadedAllPreviousMessages else { return }
 
-        // Only start fetching new messages once the user is nearing the end of the list.
-        guard indexPath.row >= self.conversationController.messages.count - 10 else { return }
-
-        self.conversationController.loadPreviousMessages()
-    }
-}
+        // Start fetching new messages once the user is nearing the end of the list.
+        guard indexPath.row < 10 else { return }
 
-extension new_ConversationViewController: SwipeableInputAccessoryViewDelegate {
-
-    func swipeableInputAccessory(_ view: SwipeableInputAccessoryView, didConfirm sendable: Sendable) {
-        if sendable.previousMessage.isNil {
-            Task { await self.send(object: sendable) }
-        } else {
-            Task { await self.update(object: sendable) }
-        }
-    }
-
-    func send(object: Sendable) async {
-        do {
-            try await self.conversationController?.createNewMessage(with: object)
-        } catch {
-            logDebug(error)
-        }
-    }
-
-    func update(object: Sendable) async {
-        do {
-            try await self.conversationController?.editMessage(with: object)
-        } catch {
-            logDebug(error)
+        Task {
+            do {
+                let oldestMessageID = self.conversationController.messages.first?.id
+                try await self.conversationController.loadPreviousMessages(before: oldestMessageID)
+            } catch {
+                logDebug(error)
+            }
         }
     }
 }
 
 extension new_ConversationViewController {
 
-    func subscribeToUpdates() {
+    func subscribeToConversationUpdates() {
         self.conversationController?.delegate = self
     }
 
-    func loadMessages() {
-        Task {
-            guard let controller = self.conversationController else { return }
+    @MainActor
```

Comment:
This has to be done on the main thread because we're changing the collectionview.